### DISCUSSION TITLE
I want to record the coordinates of feature points in an image.

### DISCUSSION THREAD

**Discussion by yyyutooo** [stella-cv/stella_vslam#147] [Q&A] [answered]
Hello.
When I run OpenVSLAM, for each frame of the image, a feature point is displayed and a point cloud appears at that location. I would like to record the coordinates of the feature points in the image, but I don't know how to change the program.
Can anyone help me? Thank you very much.

**Comment by ymd-stella** [comment, marked as answer]
It is easy to change the following line

https://github.com/OpenVSLAM-Community/openvslam/blob/214046ef4b0ffdc0784cee27ab807e4dbbce9a92/src/openvslam/publish/frame_publisher.cc#L88

If you write about why you are trying to do this, we may be able to give you more appropriate advice.

**Reply by yyyutooo** [reply]
Thank you for your answer.
I would like to record the coordinates (x-coordinates, y-coordinates) of the feature points displayed in the image for each frame, for each frame, using a CSV file or something similar. For this purpose, I would like to know which variables correspond to the x- and y-coordinates of the feature points in the image.

![Untitled 2](https://user-images.githubusercontent.com/70961161/130345519-0d2baccd-65a8-46b9-9406-3eb9db8957cc.jpg)

**Reply by ymd-stella** [reply]
I recommend you to read the following process:
https://github.com/OpenVSLAM-Community/openvslam/blob/214046ef4b0ffdc0784cee27ab807e4dbbce9a92/src/openvslam/publish/frame_publisher.cc#L76

https://github.com/OpenVSLAM-Community/openvslam/blob/214046ef4b0ffdc0784cee27ab807e4dbbce9a92/src/openvslam/publish/frame_publisher.cc#L112-L140

**Reply by yyyutooo** [reply]
I see.
I see that OpenVSLAM uses "draw_tracked_points" to draw feature points on each frame.
So "curr_keypts.at(i).pt.x" and "curr_keypts.at(i).pt.x" are the coordinates of the feature points.
Thank you very much. I'll try my best!
